Do dependencies get built as static for MinGW? In [MingW (MacOS)... shared...], I saw a lot of "multiple definition of..." from line 2854 to 3608. I'm not sure what I can do to fix this. Note that all MingW tests passed in the first commit for some reason...

Or is this because of trying to build 2 versions together?

--- a/packages/g/glibmm/xmake.lua
+++ b/packages/g/glibmm/xmake.lua
@@ -44,8 +44,7 @@ package("glibmm")
     on_install("!android and !wasm and !iphoneos and !bsd", function (package)
         -- Prevent building generate_defs_glib, generate_defs_gio. Otherwise when being built,
         -- errors arise on mingw builds. Doesn't hurt to filter for all platforms.
-        io.replace(path.join(package:sourcedir(), "tools/extra_defs_gen/meson.build"),
-            "executable(", "# executable(", {plain = true})
+        io.replace("tools/extra_defs_gen/meson.build", "executable(", "# executable(", {plain = true})
         local configs = {"-Dbuild-documentation=false", "-Dbuild-examples=false"}
         table.insert(configs, "-Dbuild-deprecated-api=" .. (package:config("deprecated_api") and "true" or "false"))
         table.insert(configs, "-Ddefault_library=" .. (package:config("shared") and "shared" or "static"))

--- a/packages/g/glibmm/xmake.lua
+++ b/packages/g/glibmm/xmake.lua
@@ -42,6 +42,10 @@ package("glibmm")
     end)
 
     on_install("!android and !wasm and !iphoneos and !bsd", function (package)
+        -- Prevent building generate_defs_glib, generate_defs_gio. Otherwise when being built,
+        -- errors arise on mingw builds. Doesn't hurt to filter for all platforms.
+        io.replace(path.join(package:sourcedir(), "tools/extra_defs_gen/meson.build"),
+            "executable(", "# executable(", {plain = true})
         local configs = {"-Dbuild-documentation=false", "-Dbuild-examples=false"}
         table.insert(configs, "-Dbuild-deprecated-api=" .. (package:config("deprecated_api") and "true" or "false"))
         table.insert(configs, "-Ddefault_library=" .. (package:config("shared") and "shared" or "static"))

--- a/packages/g/glibmm/xmake.lua
+++ b/packages/g/glibmm/xmake.lua
@@ -19,7 +19,7 @@ package("glibmm")
 
     on_load(function (package)
         -- glibmm doesn't allow static build for MSVC-like compilers.
-        if package:is_plat("windows") then
+        if package:toolchain("msvc") then
             package:config_set("shared", true)
         end
 

